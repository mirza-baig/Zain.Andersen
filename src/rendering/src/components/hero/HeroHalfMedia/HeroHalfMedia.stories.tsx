// Global
import { Story, Meta } from '@storybook/react';
// Local
import HeroHalfMedia, { HeroHalfMediaProps } from './HeroHalfMedia';
import defaultData from './HeroHalfMedia.mock-data';
import darkData from './HeroHalfMedia.mock-data-dark';
import leftAlignData from './HeroHalfMedia.mock-data-left-align';
import grayData from './HeroHalfMedia.mock-data-gray';
import videoData from './HeroHalfMedia.mock-data-video';

export default {
  title: 'Hero/HeroHalfMedia',
  component: HeroHalfMedia,
} as Meta;
const Template: Story<HeroHalfMediaProps> = (props) => <HeroHalfMedia {...props} />;

export const Default = Template.bind({});
Default.args = defaultData;

export const Dark = Template.bind({});
Dark.args = darkData;

export const LeftAlign = Template.bind({});
LeftAlign.args = leftAlignData;

export const Gray = Template.bind({});
Gray.args = grayData;

export const Video = Template.bind({});
Video.args = videoData;
