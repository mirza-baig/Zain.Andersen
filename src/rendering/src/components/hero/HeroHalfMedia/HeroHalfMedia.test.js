// Globalit('renders correctly', async () => {
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import HeroHalfMedia from './HeroHalfMedia';
import defaultData from './HeroHalfMedia.mock-data';
import darkData from './HeroHalfMedia.mock-data-dark';
import grayData from './HeroHalfMedia.mock-data-gray';
import leftAlignData from './HeroHalfMedia.mock-data-left-align';
import videoData from './HeroHalfMedia.mock-data-video';

beforeEach(() => {
  // IntersectionObserver isn't available in test environment
  const mockIntersectionObserver = jest.fn();
  mockIntersectionObserver.mockReturnValue({
    observe: () => null,
    unobserve: () => null,
    disconnect: () => null
  });
  window.IntersectionObserver = mockIntersectionObserver;
});

it('renders correctly', async () => {
  const component = snapshot(HeroHalfMedia, { componentProps: defaultData });
  await hasDataComponent(component, 'hero/herohalfmedia');
});

it('dark renders correctly', async () => {
  const component = snapshot(HeroHalfMedia, { componentProps: darkData });
  await hasDataComponent(component, 'hero/herohalfmedia');
});

it('left align renders correctly', async () => {
  const component = snapshot(HeroHalfMedia, { componentProps: leftAlignData });
  await hasDataComponent(component, 'hero/herohalfmedia');
});

it('gray renders correctly', async () => {
  const component = snapshot(HeroHalfMedia, { componentProps: grayData });
  await hasDataComponent(component, 'hero/herohalfmedia');
});

it('video renders correctly', async () => {
  const component = snapshot(HeroHalfMedia, { componentProps: videoData });
  await hasDataComponent(component, 'hero/herohalfmedia');
});
