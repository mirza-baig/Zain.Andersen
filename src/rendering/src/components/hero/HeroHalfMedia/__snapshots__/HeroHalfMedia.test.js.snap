// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dark renders correctly 1`] = `
<section
  data-component="hero/herohalfmedia"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-black bg-theme-bg"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-0 gap-x-0 "
    >
      <div
        class="col-span-12 ml:col-span-6 ml:content-right ml:items-right"
      >
        <div
          class="relative hidden md:block aspect-picture w-full  "
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              sizes="100vw"
              src="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=3840&q=75"
              srcset="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
        </div>
        <div
          class="relative block w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              sizes="100vw"
              src="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=3840&q=75"
              srcset="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
            />
          </span>
        </div>
      </div>
      <div
        class="col-span-12 ml:col-span-6 ml:flex ml:flex-col ml:self-center p-m pt-s ml:pt-m ml:px-xl xl:px-xxl ml:-order-1"
      >
        <div
          class="text-theme-text text-sm-xxs text-xxs uppercase mb-s"
        >
          <h2>
            Eyebrow
          </h2>
        </div>
        <div
          class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-l ml:text-l font-heavy mb-2 last:mb-0"
        >
          <h1>
            This is a headline
          </h1>
        </div>
        <div
          class="text-theme-body text-body font-regular mb-m last:mb-0"
        >
          <div
            class="body-copy"
          >
            <div>
              <p>
                <span>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 flex flex-col ml:flex-row items-start mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            aria-label="Call to action"
            class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed ml:mr-3 ml:first:ml-0 ml:last:mr-0 mr-m"
            data-gtm-click=""
            data-gtm-dl-event="cta_click"
            href="/Hero-Half-Media"
            role="link"
            tabindex="0"
            target=""
            title=""
          >
            Call to action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`gray renders correctly 1`] = `
<section
  data-component="hero/herohalfmedia"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-gray bg-light-gray"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-0 gap-x-0 "
    >
      <div
        class="col-span-12 ml:col-span-6 ml:content-right ml:items-right"
      >
        <div
          class="relative hidden md:block aspect-picture w-full  "
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              sizes="100vw"
              src="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=3840&q=75"
              srcset="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
        </div>
        <div
          class="relative block w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              sizes="100vw"
              src="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=3840&q=75"
              srcset="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
            />
          </span>
        </div>
      </div>
      <div
        class="col-span-12 ml:col-span-6 ml:flex ml:flex-col ml:self-center p-m pt-s ml:pt-m ml:px-xl xl:px-xxl ml:-order-1"
      >
        <div
          class="text-theme-text text-sm-xxs text-xxs uppercase mb-s"
        >
          <h2>
            Eyebrow
          </h2>
        </div>
        <div
          class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-l ml:text-l font-heavy mb-2 last:mb-0"
        >
          <h1>
            This is a headline
          </h1>
        </div>
        <div
          class="text-theme-body text-body font-regular mb-m last:mb-0"
        >
          <div
            class="body-copy"
          >
            <div>
              <p>
                <span>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 flex flex-col ml:flex-row items-start mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            aria-label="Call to action"
            class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed ml:mr-3 ml:first:ml-0 ml:last:mr-0 mr-m"
            data-gtm-click=""
            data-gtm-dl-event="cta_click"
            href="/Hero-Half-Media"
            role="link"
            tabindex="0"
            target=""
            title=""
          >
            Call to action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`left align renders correctly 1`] = `
<section
  data-component="hero/herohalfmedia"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-0 gap-x-0 "
    >
      <div
        class="col-span-12 ml:col-span-6 ml:content-right ml:items-right"
      >
        <div
          class="relative hidden md:block aspect-picture w-full  "
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              sizes="100vw"
              src="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=3840&q=75"
              srcset="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
        </div>
        <div
          class="relative block w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              sizes="100vw"
              src="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=3840&q=75"
              srcset="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
            />
          </span>
        </div>
      </div>
      <div
        class="col-span-12 ml:col-span-6 ml:flex ml:flex-col ml:self-center p-m pt-s ml:pt-m ml:px-xl xl:px-xxl"
      >
        <div
          class="text-theme-text text-sm-xxs text-xxs uppercase mb-s"
        >
          <h2>
            Eyebrow
          </h2>
        </div>
        <div
          class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-l ml:text-l font-heavy mb-2 last:mb-0"
        >
          <h1>
            This is a headline
          </h1>
        </div>
        <div
          class="text-theme-body text-body font-regular mb-m last:mb-0"
        >
          <div
            class="body-copy"
          >
            <div>
              <p>
                <span>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 flex flex-col ml:flex-row items-start mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            aria-label="Call to action"
            class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed ml:mr-3 ml:first:ml-0 ml:last:mr-0 mr-m"
            data-gtm-click=""
            data-gtm-dl-event="cta_click"
            href="/Hero-Half-Media/"
            role="link"
            tabindex="0"
            target=""
            title=""
          >
            Call to action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly 1`] = `
<section
  data-component="hero/herohalfmedia"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-0 gap-x-0 "
    >
      <div
        class="col-span-12 ml:col-span-6 ml:content-right ml:items-right"
      >
        <div
          class="relative hidden md:block aspect-picture w-full  "
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              sizes="100vw"
              src="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=3840&q=75"
              srcset="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fsmall&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
        </div>
        <div
          class="relative block w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
            />
            <img
              alt="Hero Half Media Image"
              data-nimg="responsive"
              decoding="async"
              sizes="100vw"
              src="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=3840&q=75"
              srcset="/_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=640&q=75 640w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=750&q=75 750w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=828&q=75 828w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1080&q=75 1080w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1200&q=75 1200w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=1920&q=75 1920w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=2048&q=75 2048w, /_next/image/?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F800x450%3Fmobile&w=3840&q=75 3840w"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
            />
          </span>
        </div>
      </div>
      <div
        class="col-span-12 ml:col-span-6 ml:flex ml:flex-col ml:self-center p-m pt-s ml:pt-m ml:px-xl xl:px-xxl ml:-order-1"
      >
        <div
          class="text-theme-text text-sm-xxs text-xxs uppercase mb-s"
        >
          <h2>
            Eyebrow
          </h2>
        </div>
        <div
          class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-l ml:text-l font-heavy mb-2 last:mb-0"
        >
          <h1>
            This is a headline
          </h1>
        </div>
        <div
          class="text-theme-body text-body font-regular mb-m last:mb-0"
        >
          <div
            class="body-copy"
          >
            <div>
              <p>
                <span>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 flex flex-col ml:flex-row items-start mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            aria-label="Call to action"
            class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed ml:mr-3 ml:first:ml-0 ml:last:mr-0 mr-m"
            data-gtm-click=""
            data-gtm-dl-event="cta_click"
            href="/Hero-Half-Media/"
            role="link"
            tabindex="0"
            target=""
            title=""
          >
            Call to action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`video renders correctly 1`] = `
<section
  data-component="hero/herohalfmedia"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-0 gap-x-0 "
    >
      <div
        class="col-span-12 ml:col-span-6 ml:content-right ml:items-right"
      >
        <div
          class="relative"
        >
          <div
            class=""
            style="position: relative; overflow: hidden; width: 100%; padding-top: 56.25%;"
          />
        </div>
      </div>
      <div
        class="col-span-12 ml:col-span-6 ml:flex ml:flex-col ml:self-center p-m pt-s ml:pt-m ml:px-xl xl:px-xxl ml:-order-1"
      >
        <div
          class="text-theme-text text-sm-xxs text-xxs uppercase mb-s"
        >
          <h2>
            Eyebrow
          </h2>
        </div>
        <div
          class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-l ml:text-l font-heavy mb-2 last:mb-0"
        >
          <h1>
            This is a headline
          </h1>
        </div>
        <div
          class="text-theme-body text-body font-regular mb-m last:mb-0"
        >
          <div
            class="body-copy"
          >
            <div>
              <p>
                <span>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-span-12 flex flex-col ml:flex-row items-start mb-s flex items-start md:flex-row md:items-center md:space-x-4"
        >
          <a
            aria-label="Call to action"
            class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed ml:mr-3 ml:first:ml-0 ml:last:mr-0 mr-m"
            data-gtm-click=""
            data-gtm-dl-event="cta_click"
            href="/Hero-Half-Media/"
            role="link"
            tabindex="0"
            target=""
            title=""
          >
            Call to action
            <span
              class="ml-xxs"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
